### PR TITLE
Making some code cleanups

### PR DESCRIPTION
**Pull request by marko-bekhta**
Hi @gsmet here are a few cleanups here and there. Mostly around creating empty arrays or lists with one element. This probably wouldn't be a visible performance improvement though :smiley:

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 1]
```diff
@@ -565,9 +565,6 @@ private boolean validateMetaConstraint(ValidationContext<?> validationContext, V
 				valueToValidate = valueContext.getValue( parent, metaConstraint.getLocation() );
 				valueContext.setCurrentValidatedValue( valueToValidate );
 			}
-			else {
-				valueToValidate = valueContext.getCurrentValidatedValue();
-			}
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: We could probably get rid of the variable altogether.

File: engine/src/main/java/org/hibernate/validator/internal/xml/MetaConstraintBuilder.java [hunk 2]
```diff
@@ -51,6 +51,7 @@
 	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
 
 	private static final Pattern IS_ONLY_WHITESPACE = Pattern.compile( "\\s*" );
+	private static final Class[] EMPTY_CLASSES_ARRAY = { };
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/xml/MetaConstraintBuilder.java]: Maybe keep the same notation you used before? new Class[0]?

File: test-utils/src/main/java/org/hibernate/validator/testutil/ValidationInvocationHandler.java [hunk 3]
```diff
@@ -21,6 +21,8 @@
  */
 public class ValidationInvocationHandler implements InvocationHandler {
 
+	private static final Object[] EMPTY_ARGS = { };
```
Comment (gsmet) [inline test-utils/src/main/java/org/hibernate/validator/testutil/ValidationInvocationHandler.java]: Same here.

**Comment by marko-bekhta** [comment]
Yes, I'll create a JIRA and address the suggestions.

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**